lib/storage/index_db.go

```diff
@@ -91,13 +91,17 @@ type indexDB struct {
 	// The db must be automatically recovered after that.
 	missingMetricNamesForMetricID atomic.Uint64
 
-	// minMissingTimestamp is the minimum timestamp, which is missing in the given indexDB.
+	// minMissingTimestampByKey holds the minimum timestamps by index search key,
+	// which is missing in the given indexDB.
+	// Key must be formed with marshalCommonPrefix function.
 	//
 	// This field is used at containsTimeRange() function only for the previous indexDB,
 	// since this indexDB is readonly.
 	// This field cannot be used for the current indexDB, since it may receive data
 	// with bigger timestamps at any time.
-	minMissingTimestamp atomic.Int64
+	minMissingTimestampByKey map[string]int64
```

Comment:
wdyt about moving the implementation details to a separate object `minMissingTimestampsTracker` (or whatever). Then we could have different implementation for single/cluster versions, but the same usage. This could reduce chance of getting conflicts afterwards

Reply:
We could merge this change into `single-node` branch and it'll work the same ( map with single item).